src/System.Net.Security/src/System/Net/_SslSessionsCache.cs

```diff
@@ -51,8 +53,10 @@ internal SslCredKey(byte[] thumbPrint, int allowedProtocols, EncryptionPolicy en
 
                 _HashCode ^= allowedProtocols;
                 _HashCode ^= (int)encryptionPolicy;
-                _AllowedProtocols = allowedProtocols;
+                _HashCode ^= serverMode ? 5 : 7; //TODO used a prime number here as it's a XOR. Figure out appropriate value.
+                _AllowedProtocols = allowedProtocols;               
```

Comment:
Breaking: the allowed protocols used to include the Client/Server flags. Now it doesn't.